Merge PrepareDownload and StartDownload

eseliger opened the pull request:
No noticeable perf improvement from it, and adding complexity to the code, so simplifying here.


## Test plan

CI.

eseliger committed:
Make changes a deque

That will give us changelist.Clear for free through destructor calls at the end of the processing loop iteration in the main thread.

eseliger committed:
Merge PrepareDownload and StartDownload

No noticeable perf improvement from it, and adding complexity to the code, so simplifying here.

ggilmore committed:
make changedfilegroups a local variable

ggilmore commented:
I made one small change: https://github.com/sourcegraph/p4-fusion/pull/23/commits/8a85d41ff0a0710950259fe2d2d5278e42804e17

changedFileGroups can now just be a local variable since we don't need to share it between the two functions

ggilmore reviewed (approved):
Nice! In the future, I wonder if the better way to express this idea is through `<Future<DownloadResult>>` or something similar? That way we can get rid of the need for the caller to explicitly call WaitForDownload() too.

eseliger commented:
> I made one small change: https://github.com/sourcegraph/p4-fusion/commit/8a85d41ff0a0710950259fe2d2d5278e42804e17

> changedFileGroups can now just be a local variable since we don't need to share it between the two functions

I actually fell for that before, too 😬 It's referenced from the main thread when creating the commit, so it has to stay as object-level state for now unfortunately :/

eseliger committed:
Cleanup log output (#24)

The 1.12 version of p4-fusion was less verbose which had the nice property that since we always show the last log line in the clone progress UI it would contains stats on the progress.
This reverts to that, and makes the output required for our test script opt-in using a compile time flag.

eseliger committed:
Revert "make changedfilegroups a local variable"

This reverts commit 8a85d41ff0a0710950259fe2d2d5278e42804e17.
